src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts

```diff
@@ -464,8 +464,13 @@ export class HostsComponent extends ListWithDetails implements OnDestroy, OnInit
   transformHostsData() {
     if (this.checkHostsFactsAvailable()) {
       _.forEach(this.hosts, (hostKey) => {
-        hostKey['memory_total_bytes'] = hostKey['memory_total_kb'] * 1024;
-        hostKey['raw_capacity'] = hostKey['hdd_capacity_bytes'] + hostKey['flash_capacity_bytes'];
+        hostKey['memory_total_bytes'] = !isNaN(hostKey['memory_total_kb'])
+          ? hostKey['memory_total_kb'] * 1024
```

Comment:
Why do we care about presentation stuff in the logic layer? `NaN` or `"N/A"` is not relevant at this stage.

IMHO that should be left for the template and pipes are good at this. We even have a pipe `empty` that basically converts `null` and `undefined` to `"-"`. If we need to deal with `NaN` let's add that check to that pipe and keep a consistent behaviour ("-" for null/undefined in some places, but "N/A" in others). That's the good thing of reusing helpers.

And going one step further: if this `NaN/null/undefined` seems to be a recurrent issue (I remember at least another 2 recent PRs dealing with it), why don't we append the `empty` pipe by default to the `cd-datatable` cells? That way we'll avoid having to manual check for this in every single piece of data that we display.

Additionally, is it normal to get a `NaN` here? Shouldn't we get a null or undefined?